analyzer/src/main/kotlin/managers/Pub.kt

```diff
@@ -113,6 +114,7 @@ class Pub(
     private val processedPackages = mutableListOf<String>()
     private val reader = PubCacheReader()
     private val gradleDefinitionFilesForPubDefinitionFiles = mutableMapOf<File, Set<File>>()
+    private var workingDir = File("")
```

Comment:
Please try to avoid creating such invalid empty file objects.

Reply:
Can you please suggest a correction? I'm new to Kotlin. Thanks!

Reply:
That's actually not a Kotlin-thing, you wouldn't do this in Java either. Also using `var`s is discouraged in Kotlin in favor of using `val`s where possible. It seems to me you only "need" this because you're not passing the `workingDir` as a parameter to all functions that need it. So I guess you could change the function signatures accordingly instead, so the `workingDir` is available everywhere you need it.